docs/guides/next-cookies.md

```diff
@@ -0,0 +1,232 @@
+# How to Get and Set Cookies in Next.js
+
+Every time you visit a website, servers send a small piece of information to your web browser called cookies. Each cookie is a `key=value` pair that controls when and where it's used. Cookies can help you:
+
+- Resume your session where you left
+- Remember your login credentials, preferences, and other settings
+
+The stored cookie data comes with a unique ID that servers send to the browser. Whenever a cookie is exchanged between your browser and the web server, the server reads this ID and knows what information to serve you. This way, you get a personalized web experience.
+
+## Cookies in Next.js
+
+Cookies in Next.js can be handled both on the client-side and the server-side. In this guide, you'll learn:
+
+- How to set & get cookies with Next.js
+- Build a [demo app](https://github.com/MaedahBatool/next-cookies) that'll help you set and get the value of a cookie both on the client-side and server-side
+
+![nextjs-cookies](/docs/guides/images/nextjs-cookies.png)
+
+Setting and getting cookies on the client-side with Next.js is exactly the same as you would do with any other web application. However, on the server-side, Next.js helps you one step further to set and get cookies (more on this later).
+
+## Client-side Cookies in Next.js
+
+You set cookies via,
+
+```js
+document.cookie = 'name=value'
+```
+
+You get cookies via,
+
+```js
+document.cookie
+```
+
+Reading `document.cookie` shows all the available cookies in the current context, where a semicolon separates each cookie.
+
+The `document.cookie` property takes up different attributes. E.g. `max-age` lets you define when the cookie should expire, `secure` allows cookies to transmit only over a secure protocol such as HTTPS.
+
+You can read more details about the `document.cookie` property on the [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie).
+
+### Set and Get Cookies on the Server-side on the Client-side
+
+Let's build a demo application to set a `theme` cookie to save the `light` or `dark` mode of your website.
+
+![client-side-cookies](/docs/guides/images/client-cookies.png)
+
+Let's create a Next.js page that shows the current value of the cookie`theme` and allows you to change it to `light` or `dark`.
+
+```js
```

Comment:
This codeblock has the `changeTheme` function, without an explanation of what it is. Maybe we could just use the second codeblock with the full code instead?